Addressing modes

rudy-pants opened the pull request:
Function to do addressing modes for any instruction

rudy-pants committed:
Addressing mode

rudy-pants committed:
Updated cpu.js with Uint8Arrays for byte storage

rudy-pants committed:
Creating memory class

lucasscharenbroch committed:
Created MemoryMap class

added MemoryMap class and its tester functions

rudy-pants committed:
Create cpu.ts

lucasscharenbroch committed:
Added getOperandAddress method

+ testers

lucasscharenbroch committed:
Create test_page.html

rudy-pants committed:
Removing comments

lucasscharenbroch committed:
Merge pull request #14 from rudyb2001/implement-memory-struct

Implement memory struct

lucasscharenbroch committed:
Add Opcode struct and ADC

ADC and basic Opcode functionality still needs to be thoroughly tested

lucasscharenbroch committed:
Update opcode.js

corrected typo

lucasscharenbroch committed:
Converted .js file to .ts, added types to most functions

lucasscharenbroch committed:
Added Opcode Testers

Added testers for the opcode class, general-use opcode functions, ADC, CLC, and MemoryMap's getOperand.
A few testing helpers were also added.

lucasscharenbroch commented:
We should be ready to start writing opcodes now. How should we go about merging into branches with the old rust/ js files?

rudyb2001 commented:
Are you talking about restructuring the file system? I think that we can get rid of all the rust files. We will have to go do some merging at some point, so I think we should just merge all of our changes now and then try the restructuring after that.

lucasscharenbroch committed:
Expanded Opcode Functionality

- added stack pointer to CPU class
- added Accumulator addressing mode
- added isWriteInstruction instance field to Opcode class, along with a new branch in Opcode.executeCurrent to handle write instructions
- added testers for several opcodes
- implemented a variety of opcodes

lucasscharenbroch commented:
Sounds good to me.

lucasscharenbroch committed:
More Opcode Adjstuments

Added a few more opcodes, testers, and helper functions; tweaked the status bit-field to align more with the hardware (added a second bit of the b-flag https://www.nesdev.org/wiki/Status_flags).
